compiler/src/iree/compiler/Codegen/Common/GPU/GPUReduceBankConflicts.cpp

```diff
@@ -17,7 +17,12 @@ namespace iree_compiler {
 /// have bank conflicts when reading 2D shapes within shared memory.
 static void padAlloc(MLIRContext *context, memref::AllocOp allocOp,
                      int64_t paddingSizeBits) {
-  int64_t innerDim = allocOp.getType().getShape().back();
+  if (!llvm::isa<ShapedType>(allocOp.getType()))
```

Comment:
nit: memref::AllocOp can only return a memref type so no need to check for a shaped type here.